tests/integration/test_convolve.py

```diff
@@ -82,6 +88,39 @@ def test_int(shape, filter_shape):
     check_convolve(a, v)
 
 
+def test_dtype():
+    shape = (5,) * 2
+    arr1 = num.random.randint(0, 5, shape)
+    arr2 = num.random.random(shape)
+    out_num = num.convolve(arr1, arr2, mode="same")
+    out_np = np.convolve(arr1, arr2, mode="same")
+    assert allclose(out_num, out_np)
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize("mode", MODES)
+def test_modes(mode):
+    shape = (5,) * 2
+    arr1 = num.random.random(shape)
+    arr2 = num.random.random(shape)
+    out_num = num.convolve(arr1, arr2, mode=mode)
+    # when mode!="same", cunumeric raises
+    # NotImplementedError: Need to implement other convolution modes
+    out_np = np.convolve(arr1, arr2, mode=mode)
+    assert allclose(out_num, out_np)
+
+
+@pytest.mark.xfail
+def test_ndim():
+    shape = (5,) * 4
+    arr1 = num.random.random(shape)
+    arr2 = num.random.random(shape)
+    out_num = num.convolve(arr1, arr2, mode="same")
+    # cunumeric raises,  NotImplementedError: 4-D arrays are not yet supported
+    out_np = np.convolve(arr1, arr2, mode="same")
+    assert allclose(out_num, out_np)
```

Comment:
This test could be improved by parameterizing different dimension cases (with an xfail param for N=4) Alternatively we could have separate tests for each N up to 4 or so.

Reply:
I would expect this exact same test to get run (without xfail) for the working case somehow

Reply:
Maybe this and the above cases would be simpler to by having a single test run all the cases (both working and non-working) and specifying the `xfail` on a per-parameter basis, e.g. like some other tests do:
```
"equal_nan", (
    False, 
    pytest.param(True, marks=pytest.mark.xfail)
)
```